internal/messagequeue/messagequeue.go

```diff
@@ -413,6 +415,9 @@ func (mq *MessageQueue) sendMessage() {
 
 			mq.simulateDontHaveWithTimeout(message)
 
+			// Signal that a request was sent
+			mq.onRequestSent(mq.p)
```

Comment:
Concerned that doing this after sending the request could cause us to handle this after receiving the response.